src/optimizers/adam.cu

```diff
@@ -131,7 +132,7 @@ void adam<TensorDataType>::step_compute_gpu(AbsDistMatrixType& values,
                                 m_eps,
                                 m_beta1,
                                 m_beta2,
-                                m_adamw_weight_decay * lr,
+                                adjusted_weight_decay,
```

Comment:
@tbennun Is this supposed to be in this PR?

Reply:
Refactored to #2337